math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala

```diff
@@ -52,81 +65,105 @@ trait CanTraverseValues[From, A] {
-    def zeros(numZero: Int, zeroValue: A)
+
+    def zeros(numZero: Int, zeroValue: V): Unit
+
   }
 
   //
   // Arrays
   //
 
-  class OpArray[@specialized(Double, Int, Float, Long) A]
-    extends CanTraverseValues[Array[A], A] {
-    /** Traverses all values from the given collection. */
-    def traverse(from: Array[A], fn: ValuesVisitor[A]): Unit = {
+  class OpArray[@specialized(Double, Int, Float, Long) V]
+    extends CanTraverseValues[Array[V], V] {
+
+
+    def traverse(from: Array[V], fn: ValuesVisitor[V]): Unit = {
       fn.visitArray(from)
     }
 
-    def isTraversableAgain(from: Array[A]): Boolean = true
+    def isTraversableAgain(from: Array[V]): Boolean = true
+
   }
 
+  implicit def opArray[@specialized A] = new OpArray[A]
+
+  /** Implicit object to apply breeze traversable operations to Array[Int].*/
+  implicit object ImplicitOpArrayInt extends OpArray[Int]
 
-  implicit def opArray[@specialized A] =
-    new OpArray[A]
+  /** Implicit object to apply breeze traversable operations to Array[Short].*/
+  implicit object ImplicitOpArrayShort extends OpArray[Short]
 
-  implicit object OpArrayII extends OpArray[Int]
+  /** Implicit object to apply breeze traversable operations to Array[Short].*/
+  implicit object ImplicitOpArrayLong extends OpArray[Long]
 
-  implicit object OpArraySS extends OpArray[Short]
+  /** Implicit object to apply breeze traversable operations to Array[Float].*/
+  implicit object ImplicitOpArrayFloat extends OpArray[Float]
 
-  implicit object OpArrayLL extends OpArray[Long]
+  /** Implicit object to apply breeze traversable operations to Array[Double].*/
+  implicit object ImplicitOpArrayDouble extends OpArray[Double]
 
-  implicit object OpArrayFF extends OpArray[Float]
+  /** Implicit object to apply breeze traversable operations to Array[Complex].*/
+  implicit object ImplicitOpArrayComplex extends OpArray[Complex]
 
-  implicit object OpArrayDD extends OpArray[Double]
+  /**
+    * Implicit conversion to apply breeze traversing operations to
+    * scala collections or iterators ([[scala.TraversableOnce]] objects).
+    */
 
-  implicit object OpArrayCC extends OpArray[Complex]
+  /**Implicit implementation which allows traversal of
+    * scala collection objects implementing [[scala.collection.TraversableOnce]]. */
   implicit def canTraverseTraversable[V,X <: TraversableOnce[V]]: CanTraverseValues[X, V] = {
     new CanTraverseValues[X, V] {
+
       /** Traverses all values from the given collection. */
       override def traverse(from: X, fn: CanTraverseValues.ValuesVisitor[V]): Unit = {
         for(v <- from) {
           fn.visit(v)
         }
       }
 
-      def isTraversableAgain(from: X): Boolean = from.isTraversableAgain
+      override def isTraversableAgain(from: X): Boolean = from.isTraversableAgain
+
     }
   }
 }
 
-trait LowPrioCanTraverseValues { this: CanTraverseValues.type =>
+@deprecated("1.0")
+trait LowPrioCanTraverseValues {
+  this: CanTraverseValues.type =>
```

Comment:
huh, yeah, CanTraverseValues is supposed to inherit from it, but obviously i missde that.